internal/controllers/analyzer/analyzer.go

```diff
@@ -457,3 +467,59 @@ func (a *Analyzer) isWarning(err string) bool {
 		strings.Contains(err, messages.MsgWarnBrakemanNotRubyOnRailsProject) ||
 		strings.Contains(err, messages.MsgWarnGemfileIsRequiredForBundler)
 }
+
+// getAllVulnerabilitiesWithDetailsJoined will add the default separator of the details and will check if the hash
+// already exists in list of the vulnerabilities, if is duplicated, so, it will only join both details.
+// nolint:funlen // This method cannot broken in other for not be confused.
+func (a *Analyzer) getAllVulnerabilitiesWithDetailsJoined(
+	detailsHeaderText string,
+) (newAnalysisVulnerabilities []analysis.AnalysisVulnerabilities) {
+	for indexAV := range a.analysis.AnalysisVulnerabilities {
+		exists := false
+		for index := range newAnalysisVulnerabilities {
+			if a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.VulnHash ==
+				newAnalysisVulnerabilities[index].Vulnerability.VulnHash {
+				exists = true
+				newAnalysisVulnerabilities[index].Vulnerability.Details = fmt.Sprintf("%s\n         %s%s",
+					newAnalysisVulnerabilities[index].Vulnerability.Details,
+					detailsHeaderText,
+					a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details)
+				break
+			}
+		}
+		if !exists {
+			a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details = fmt.Sprintf("%s%s",
+				detailsHeaderText, a.analysis.AnalysisVulnerabilities[indexAV].Vulnerability.Details)
+			newAnalysisVulnerabilities = append(newAnalysisVulnerabilities, a.analysis.AnalysisVulnerabilities[indexAV])
+		}
+	}
+	return newAnalysisVulnerabilities
```

Comment:
Since the `newAnalysisVulnerabilities` is a named result parameter, this can simplified to `return`

Reply:
done :)